src/main/scala/chiseltest/backends/SimulatorInterface.scala

```diff
@@ -1,30 +1,46 @@
+// SPDX-License-Identifier: Apache-2.0
+
 package chiseltest.backends
 
-import firrtl.AnnotationSeq
+import chisel3.Data
+import firrtl.annotations.NoTargetAnnotation
+import logger.LazyLogging
+
+/** annotation to store [[SimulatorInterface]], which will be used in [[chiseltest.internal.ThreadedBackend]]. */
+case class SimulatorInterfaceAnnotation(interface: SimulatorInterface) extends NoTargetAnnotation
 
-/** API to Simulator. */
-trait SimulatorInterface {
+/** Standard API which should be implemented by a [[SimulatorInterface]], */
+trait SimulatorInterface extends LazyLogging {
 
   /** start time of this test. */
-  final val startTime: Long = System.nanoTime()
+  final private[chiseltest] lazy val startTime: Long = System.nanoTime()
 
-  /** how many nano seconds has elapsed since this test start. */
-  def elapsedNanoSeconds(): Long = System.nanoTime() - startTime
+  /** start time of this test. */
+  final private[chiseltest] lazy val endTime: Long = elapsedNanoSeconds()
 
-  def annotations: AnnotationSeq
+  /** how many nano seconds has elapsed since this test start. */
+  final private[chiseltest] def elapsedNanoSeconds(): Long = System.nanoTime() - startTime
 
   /** poke a signal name with a [[BigInt]]. */
-  def poke(signal: String, value: BigInt): Unit
+  private[chiseltest] def poke(signal: String, value: BigInt): Unit
+
+  /** peek a signal name, return BigInt if found.
+    * If signal is not found in circuit, return [[None]].
+    */
+  private[chiseltest] def peek(signal: String): Option[BigInt]
 
-  /** peek a signal name, return a BigInt if found. */
-  def peek(signal: String): Option[BigInt]
+  /** Ask Simulator to step to next n delta cycle. */
```

Comment:
Does it make sense to define delta cycles? What does it even mean to step multiple delta cycles? As far as I understand, the current way things work is that there's an update that propagates all combinational operations - and there isn't really a concept of how many updates there is, it's updated until essentially fixed-point?